frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/ToggleListingCommitsAction.java

```diff
@@ -38,16 +39,30 @@ public void update(AnActionEvent anActionEvent) {
     if (branchLayout.isDefined()) {
       boolean anyChildBranchExists = branchLayout.get().getRootEntries()
           .exists(rootBranch -> rootBranch.getChildren().nonEmpty());
+
       if (anyChildBranchExists) {
-        presentation.setEnabled(true);
-        presentation.setDescription("Toggle listing commits");
+        boolean anyCommitExists = getGitMacheteRepositorySnapshot(anActionEvent)
```

Comment:
Nice, I noticed that some time ago as well... not sure if I opened any issue though